treeherder/etl/perf.py

```diff
@@ -176,9 +176,11 @@ def _load_perf_datum(job: Job, perf_datum: dict):
             suite_extra_options = _order_and_concat(suite['extraOptions'])
         summary_signature_hash = None
 
-        # if we have a summary value, create or get its signature by all its subtest
-        # properties.
-        if suite.get('value') is not None:
+        # If we have a summary value, create or get its signature by all its subtest
+        # properties. For benchmarks, we need to get a summary signature hash so that
+        # the series doesn't reset when we start moving away from the summary value,
+        # and start including that value as it's own subtest with replicates.
+        if suite.get('value') is not None or suite.get("type") == "benchmark":
```

Comment:
I am wondering if it's possible to have summary signature hash for a benchmark test with a suite-level value. After removing the value from a benchmark isn't the function `_get_signature_hash(summary_properties)` going to generate a different hash (because of the missing value)? and as a consequence another signature will be created?

Reply:
No it won't create an additional one because the value is never added to the `summary_properties` dict so the hash isn't based on it.